test/metabase/db/schema_migrations_test.clj

```diff
@@ -1494,10 +1496,13 @@
 
 (deftest cache-config-migration-test
   (testing "Caching config is correctly copied over"
-    (impl/test-migrations "v50.2024-04-12T12:33:09" [migrate!]
-      (mdb.query/update-or-insert! :model/Setting {:key "enable-query-caching"} (constantly {:value "true"}))
-      (mdb.query/update-or-insert! :model/Setting {:key "query-caching-ttl-ratio"} (constantly {:value "100"}))
-      (mdb.query/update-or-insert! :model/Setting {:key "query-caching-min-ttl"} (constantly {:value "123"}))
+    (impl/test-migrations ["v50.2024-06-12T12:33:07"] [migrate!]
+      ;; this peculiar setup is to reproduce #44012, `enable-query-caching` should be unencrypted for the condition
+      ;; to hit it
+      (t2/insert! :setting [{:key "enable-query-caching", :value (encryption/maybe-encrypt "true")}])
```

Comment:
If the test instance has encryption enabled, this will still encrypt the value, but with the encryption key for the instance. Just remove the `maybe-encrypt` call entirely like this:
```suggestion
      (t2/insert! :setting [{:key "enable-query-caching", :value "true"}])
```